### PR TITLE
fix bug in readBool and ReadUInt8

### PR DESCRIPTION
**Pull request by SaitTalhaNisanci**

### CODE REVIEW

File: internal/common/buffer_util_test.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func TestReadBool(t *testing.T) {
 	buf := []byte{0, 0, 0}
 	WriteBool(buf, 0, true)
 	WriteBool(buf, 2, true)
-	if ReadBool(buf, 0) != true && ReadBool(buf, 2) != true {
+	if ReadBool(buf, 0) != true || ReadBool(buf, 2) != true {
```
Comment (furkansenharputlu) [inline internal/common/buffer_util_test.go]: This should be `OR`. No need to change.

File: internal/common/buffer_util_test.go [hunk 2]
```diff
@@ -67,7 +67,7 @@ func TestReadUInt8(t *testing.T) {
 	buf := []byte{0, 0, 0}
 	WriteUInt8(buf, 1, 5)
 	WriteUInt8(buf, 2, 12)
-	if ReadUInt8(buf, 1) != 5 && ReadUInt8(buf, 2) != 12 {
+	if ReadUInt8(buf, 1) != 5 || ReadUInt8(buf, 2) != 12 {
```
Comment (furkansenharputlu) [inline internal/common/buffer_util_test.go]: same

File: internal/common/buffer_util_test.go (hunk 1 above)
Comment (furkansenharputlu) [reply internal/common/buffer_util_test.go]: If one of them is not `true`. It should fail.

Comment (SaitTalhaNisanci) [reply internal/common/buffer_util_test.go]: That is why it should be 'or'. Currently it is 'and' so if one of them fails it doesnt fail

Comment (furkansenharputlu) [reply internal/common/buffer_util_test.go]: I am confused. Ignore this.

**Pull request merged**